core/src/main/scala/org/apache/spark/util/Utils.scala

```diff
@@ -1976,6 +1976,40 @@ private[spark] object Utils extends Logging {
     }
   }
 
+  /**
+   * Returns a qualified path object, API ported from org.apache.hadoop.fs.Path class
+   * @param defaultUri default uri corresponding to the filesystem provided.
+   * @param workingDir the working directory for the particular child path wd-relative names.
+   * @param path Path instance based on the path string specified by the user.
+   * @return Returns a qualified path object
+   */
+  def makeQualified(defaultUri: URI, workingDir: Path, path: Path): Path = {
+    val pathUri = if (!path.isAbsolute()) new Path(workingDir, path).toUri() else path.toUri()
+    val newPaths = {
```

Comment:
I think this brace isn't needed? In fact the var `newPaths` isn't needed? would be `newPath` I think anyway.